src/lib/comms/sol-http-server-impl-microhttpd.c

```diff
@@ -231,12 +236,33 @@ headers_iterator(void *data, enum MHD_ValueKind kind, const char *key, const cha
 }
 
 static int
+get_static_file(const char *dir, const char *url)
+{
+    int ret;
+    char path[PATH_MAX];
+
+    /* url given by microhttpd starts from /. For instance,
+     * https://www.solettaproject.com => url == /
+     * https://www.solettaproject.com/thankyou => url == /thankyou
+     */
+    ret = snprintf(path, sizeof(path), "%s/%s", dir,
```

Comment:
It's also important to return only a simple "Not found" error in the case that the prefixes don't match. This way, only one error will be returned, avoiding people trying to discover information about the system by testing if the server returns different errors (such as "Access denied") from different paths.